websocket.go

```diff
@@ -614,6 +615,10 @@ func (c *wsConn) handleWsConn(ctx context.Context) {
 					c.handleFrame(ctx, frame)
 					go c.nextMessage()
 					continue
+				} else {
+					err = fmt.Errorf("got err while decoder reader %w", err)
+					c.incomingErr = err //must set incomingErr, tell the request fail immediately
+					close(c.incoming)
```

Comment:
If we want to drop the connection, we should just call `c.conn.Close` and return.

Reply:
if ignore this error. a request may send to to unstable connection, but never get a response 

1.  got decode error and  ```tryReconnect```. (inflight was removed) but not success
2.  a request send before tryReconnect success(conn not change to new one), send can success because the network was not comlete unreachable
3. tryReconnect success, and change a new conn, but the request in step 2 will never get response
4. client request get stuck for ever